feat(css): add baseline-source prop (#42228)
mdn/content: files/en-us/web/css/reference/properties/baseline-source/index.md

---
title: baseline-source
slug: Web/CSS/Reference/Properties/baseline-source
page-type: css-property
browser-compat: css.properties.baseline-source
sidebar: cssref
---

The **`baseline-source`** [CSS](/en-US/docs/Web/CSS) property defines which [baseline](/en-US/docs/Web/CSS/Reference/Values/baseline-position) to use when inline-level boxes have multiple possible baselines, such as multi-line inline blocks or inline flex containers.
The values allow for choosing between aligning to the box's first baseline, last baseline, or letting the browser decide automatically based on the box type.

## Syntax

```css
/* Keyword values */
baseline-source: auto;
baseline-source: first;
baseline-source: last;

/* Global values */
baseline-source: inherit;
baseline-source: initial;
baseline-source: revert;
baseline-source: revert-layer;
baseline-source: unset;
```

> The `baseline-source` property is specified as one of the keywords listed below.

Reviewer: We should likely make how we intro values uniform across all properties. Should we ask @dipikabh?

(from `forced-color-adjust`)
```suggestion
The `baseline-source` property's value must be one of the following keywords.
```

or (from `clear`)
```suggestion
```

or (from `hypens`)

```suggestion
The `baseline-source` property is specified as a single keyword value chosen from the list below.
```

my preference is the empty `clear` version

### Values

- `auto`

>   Specifies [last-baseline](/en-US/docs/Web/CSS/Reference/Values/baseline-position#last_baseline) alignment for inline-block, [first-baseline](/en-US/docs/Web/CSS/Reference/Values/baseline-position#first_baseline) alignment for everything else.

Author: done in e92690dfc6f4d63c47e6f93e5f7dfa937c1c827f

In the following example, the inline flex container on the left uses the `first` baseline, `auto` uses the default baseline (`first` for inline flex containers), and the one on the right uses the `last` baseline.

#### HTML

```html
<span>Baseline ___</span>

<span class="box first">
  <span>First</span>
  <span>A</span>
  <span>B</span>
  <span>C</span>
</span>

<span class="box auto">
  <span>Auto</span>
  <span>A</span>
  <span>B</span>
  <span>C</span>
</span>

<span class="box last">
  <span>A</span>
  <span>B</span>
  <span>C</span>
  <span>Last</span>
</span>
```

#### CSS

```css hidden
body {
  font-family: sans-serif;
}

.box {
  display: inline-flex;
  flex-direction: column;
  border: 2px solid #888;
  width: 50px;
}

span {
  padding: 0.4rem;
}
```

```css
.first {
  baseline-source: first;
}

.last {
  baseline-source: last;
}

.auto {
  baseline-source: auto;
}
```

#### Result

{{EmbedLiveSample('baseline_selection_in_inline_flex_containers', '100%', 260)}}

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

- [`vertical-align`](/en-US/docs/Web/CSS/Reference/Properties/vertical-align) property

> - [CSS box alignment](/en-US/docs/Web/CSS/Guides/Box_alignment/Overview)

Reviewer: ```suggestion
- [CSS box alignment overview](/en-US/docs/Web/CSS/Guides/Box_alignment/Overview)
```
